### PR TITLE
(RE-8726) Allow non-final repos to be set and used

### PR DESCRIPTION
**Pull request by melissa**

### CODE REVIEW

**Comment by mckern** [comment]
What's a non-final repo?

**Comment by melissa** [comment]
@mckern get outta here!

**Comment by mckern** [comment]
Why? I said I'd stick around on this project for funsies. What's a non-final repo?

**Comment by melissa** [comment]
Beta, nightlies, development. We want to be able to ship continuously eventually, and have the automation distinguish between what can go in the final repos, and what can go into the non-final/beta/pre-release/development/nightly repos. I'll be sure to add a way better description when I clean up the commits

File: Gemfile [hunk 1]
```diff
@@ -22,11 +22,14 @@ end
...
+  gem 'packaging', *location_for(ENV['PACKAGING_LOCATION'] || '0.4.3')
   gem 'rake', require: false
   gem 'rspec', '~> 3.0', require: false
   gem 'rubocop', "~> 0.47", require: false
   gem 'simplecov', require: false
   gem 'yard', require: false
 end
+
+gem 'pry'
+gem 'pry-nav'
```
Comment (demophoon) [inline Gemfile]: Just making a note here to remind you to remove this before merging

**Comment by mckern** [comment]
> have the automation distinguish between what can go in the final repos, and what can go into the non-final/beta/pre-release/development/nightly repos

Which automation?

**Comment by melissa** [comment]
@mckern the build automation, so either the packaging repo or vanagon, depending on what's doing the actual building. We need to stage packages in the repo they're going to end up in. In the case of puppet5, that'll either be the puppet5 repo or the puppet5-nightly repo. We need to do the same switch with the keys, so we sign final packages with our main signing key, and we sign development/nightly/beta packages with the nightly key.

**Comment by mckern** [comment]
OK. I'll stay out of your hair until you've got some tests written and you're ready for review. Thanks for filling in the blanks for me!

**Comment by melissa** [comment]
We're doing this a different way

**Pull request closed without merging**